trending_data/src/main/webapp/script.js

```diff
@@ -12,7 +13,7 @@ function initMap() {
     zoom: 3,
     minZoom: 2,
   });
-
+  infoWindws = new InfoWindowSingleton(map);
```

Comment:
nit: you could call this variable something more suggestive e.g. **infoWindowManager** or **windowSingleton**

Reply:
changed name to UniqueWindowHandler, is it better now?

Reply:
I was referring to the variable name not the class name. So instead of infoWindws you could say  **infoWindowManager** or **windowSingleton**. The class name is great